internal/credential/store_test.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package credential
+
+import (
+	"os"
+	"reflect"
+	"testing"
+)
+
+func TestNewStoreError(t *testing.T) {
+	filename := "testfile.txt"
+	_, err := os.Create(filename)
+	if err != nil {
+		t.Errorf("error while creating test file %v", err)
+	}
+	defer func() {
+		err := os.Remove(filename)
+		if err != nil {
+			t.Errorf("error while removing test file %v", err)
+		}
+	}()
```

Comment:
There is a better way to create test files for testing.

Could you take a look at [`t.TempDir()`](https://pkg.go.dev/testing#T.TempDir) and modify the code accordingly? Basically, it will do the clean up automatically for you.

Reply:
Yes, I'll check this.